batchflow/notifier.py

```diff
@@ -0,0 +1,222 @@
+""" Progress notyfier. """
+import math
+from time import time
+
+from tqdm import tqdm
+from tqdm.notebook import tqdm as tqdm_notebook
+from tqdm.auto import tqdm as tqdm_auto
+
+import numpy as np
+from IPython import display
+import matplotlib.pyplot as plt
+
+from .monitor import ResourceMonitor, MONITOR_ALIASES
+from .named_expr import NamedExpression, eval_expr
+
+class Notifier:
```

Comment:
Let's make a bit more structured class hierarchy like:
- Notifier
---- Bar
---- Monitor
------ GPU_Monitor
etc

and Notifer can also take a list of other notifiers and update them at once

Reply:
What is the reason for it? Currently, `ResourceMonitor` has nothing to do with notifications and the class is used to generate some data, while `Notifier` is used to display it, so that class hierarchy would be rather strange.

As for the `Notifier` being able to initialize from a list of previously created `Notifier`s: should it be a creation of a new instance that tracks the same monitors/pipeline variables or just storing all the passed notifiers and clicking `update` on them at the same time when `self` is `update`d? While the second option is very easy to implement, what is the use case for it? The first option is way harder and would require making some decisions; it would also make the code hard to read: the main problem is what to do with `bar`s from each from the passed `Notifier`. Considering the fact that bar creation is already the hardest part of the proposed code, in either way it would render a (currently) very simple and clear class completely unreadable.

With all that, my question is: what is the benefit/use case of going all this trouble?